src/flag_gems/ops/cross_entropy_loss.py

```diff
@@ -198,18 +201,29 @@ def forward(ctx, input, target, weight, reduction, ignore_index, label_smoothing
         logging.debug("GEMS CrossEntropyLoss")
         assert reduction in Reduction._value2member_map_, "Invalid reduction"
         assert isinstance(input, torch.Tensor), "input is not a tensor"
+
         if input.ndim >= 2:
             dim = 1
         else:
             dim = 0
         if reduction != Reduction.MEAN.value:
             mean_num = -1
         else:
-            mean_num = -target.numel()
+            if ignore_index != -100:
```

Comment:
It occurs to me -100 is used as the default ignore_index value that indicates that the target class for a particular input is to be ignored.  Here the targets are ignored only when ignore_index is not -100.

Reply:
it suffices to remove this condition to make it correct.